airflow/models/taskinstance.py

```diff
@@ -1746,7 +1745,8 @@ def handle_failure(
 
         if error:
             if isinstance(error, BaseException):
-                execution_frame = _EXECUTION_FRAME_MAPPING.get(self.task)
+                execution_frame = _LAST_FRAME_LOCAL_STORAGE.frame
+                _LAST_FRAME_LOCAL_STORAGE.frame = None
```

Comment:
I think `del` is nicer because then it will fail if there is a logical error in the code.

Reply:
Aren't you afraid of some edge cases here @malthe ? I think  Iprefer to have a logical error handled nicely here as this is not a critical stuff (it will work just fine). Maybe we should instead issue WARNING if it is not set here?

Reply:
Yes we can do that:
```python
if _LAST_FRAME_LOCAL_STORAGE.__dict__.pop("frame", None) is None:
    warn(...)
```

Reply:
Yeah. I optimized the code a little bit and did it a bit differently (more info). One minute.

Reply:
See now @malthe